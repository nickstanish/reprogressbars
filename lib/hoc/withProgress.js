import React from 'react';
import PropTypes from 'prop-types';
import ProgressProvider from '../components/ProgressProvider';

export default function withProgress(ProgressBarComponent) {
  function ProgressBarProvider(props) {
    return React.createElement(
      ProgressProvider,
      { isLoading: props.isLoading },
      React.createElement(ProgressBarComponent, null)
    );
  }

  ProgressBarProvider.propTypes = {
    isLoading: PropTypes.bool
  };

  return ProgressBarProvider;
}