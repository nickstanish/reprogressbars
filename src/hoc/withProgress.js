import React, { PropTypes } from 'react';
import ProgressProvider from '../components/ProgressProvider';

export default function withProgress(ProgressBarComponent) {
  function ProgressBarProvider(props) {
    return (
      <ProgressProvider isLoading={props.isLoading}>
        <ProgressBarComponent />
      </ProgressProvider>
    );
  }

  ProgressBarProvider.propTypes = {
    isLoading: PropTypes.bool
  };

  return ProgressBarProvider;
}
