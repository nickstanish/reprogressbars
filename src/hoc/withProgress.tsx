import React from "react";
import { ProgressProvider } from "../components/ProgressProvider";
import { ProgressState } from "../components/types";

export interface WithProgressProps {
  isLoading: boolean;
}

interface InjectedProps {
  readonly progress: ProgressState;
}

export function withProgress<ComponentProps>(
  WrappedComponent: React.ComponentType<ComponentProps & InjectedProps>
): React.ComponentType<ComponentProps & WithProgressProps> {
  const WrappedProgressBar: React.FunctionComponent<
    ComponentProps & WithProgressProps
  > = (props) => {
    return (
      <ProgressProvider
        isLoading={props.isLoading}
        render={({ progress }) => {
          return <WrappedComponent {...props} progress={progress} />;
        }}
      />
    );
  };
  return WrappedProgressBar;
}
