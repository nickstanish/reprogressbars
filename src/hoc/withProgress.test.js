/* eslint-disable react/prop-types */
import React from 'react';
import { mount } from 'enzyme';

import withProgress from './withProgress';

function Target(props) {
  return (
    <p>{props.progress.value}</p>
  );
}

describe('withProgress', () => {
  test('wraps progress provider', () => {
    const TargetWithProgress = withProgress(Target);
    expect(mount(<TargetWithProgress />)).toMatchSnapshot();
  });
});

