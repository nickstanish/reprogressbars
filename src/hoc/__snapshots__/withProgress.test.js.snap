// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withProgress wraps progress provider 1`] = `
<ProgressBarProvider>
  <ProgressProvider>
    <Target
      progress={
        Object {
          "active": false,
          "value": 0,
        }
      }
    >
      <p>
        0
      </p>
    </Target>
  </ProgressProvider>
</ProgressBarProvider>
`;
